### DISCUSSION TITLE
"Login to Outline"

### DISCUSSION THREAD

**Discussion by hberg539** [outline/outline#2882] [General] [answered]
I'm trying to self-host outline on my dedicated server and the inital setup worked great. This is my start command:
`docker run --env-file=outline-env --net="host" outlinewiki/outline`
`yarn run v1.22.15
$ node ./build/server/index.js
{"label":"lifecycle","level":"info","message":"Note: Restricting process count to 1 due to use of collaborative service"}
{"label":"lifecycle","level":"info","message":"\nIs your team enjoying Outline? Consider supporting future development by sponsoring the project:\n\nhttps://github.com/sponsors/outline\n"}
{"label":"lifecycle","level":"info","message":"Note: Restricting process count to 1 due to use of collaborative service"}
{"label":"lifecycle","level":"info","message":"Starting collaboration service"}
{"label":"lifecycle","level":"info","message":"Starting websockets service"}
{"label":"lifecycle","level":"info","message":"Starting worker service"}
{"label":"lifecycle","level":"info","message":"Starting web service"}
{"level":"warn","message":"Enforced https was disabled with FORCE_HTTPS env variable"}
{"label":"lifecycle","level":"info","message":"Listening on http://localhost:3000"}
{"label":"http","level":"info","message":"  <-- GET /"}
`

I've allowed the port 3000 on the firewall, but when i open the browser, i only get the following output:
![grafik](https://user-images.githubusercontent.com/902223/147406554-dae05dc1-b8f8-41c1-bf99-43c49f45fbea.png)

No error on the browser console and no error on the console output. What i'm doing wrong?
Thank you.

**Comment by hberg539** [comment, marked as answer]
Solved it. There were not authentication methods enabled.

**Reply by ovizii** [reply]
> Solved it. There were not authentication methods enabled.

Thanks for updating your discussion. Does this mean we have to use one of the 3 authentication methods listed: slack, google or microsoft? And if the answer is yes, is there a method to just test outline locally? As far as I understand, using any of these 3 methods means I need to provide a callback URL for outline, meaning I can't just test it locally without being reachable via an external FQDN?

**Reply by airtonix** [reply]
> User never replies.

**Comment by tonick2001** [comment]
Hello. Did you manage to solve the problem?
Here there is about email authorization, but something doesn't work either!
https://wiki.generaloutline.com/s/hosting/doc/email-magic-link-N2CPh5tmTS

**Reply by ovizii** [reply]
I just tried again, and I can't get the email authorization to work either. I have defined SMTP_HOST, SMTP_PORT, SMTP_FROM_EMAIL correctly but it seems we are missing some ENV variable or config to allow login with email link. 

The instructions are slightly ambiguous, at first it says its enabled by default then it says you have to “Allow email authentication” under settings which we can only reach once we are authorized.

```
Configuration

Email authentication is enabled by default if you have correctly setup an [SMTP provider](https://github.com/doc/smtp-cqCJyZGMIB) to send emails from the app.

You must also ensure the "Allow email authentication” is enabled under Settings → Security, it should look like the following:

```

**Reply by tommoor** [reply]
There needs to be an existing team for email auth, there isn't a way to create one outside of the SSO sign-up I'm afraid. It just hasn't been built.

**Reply by ovizii** [reply]
I jsut gave it a try: I enabled auth with slack, logged in, enabled authentication with email, logged out and indeed I can chose between slack and email but when using email, after I type in my email address and hit enter, it continues through slack auth.

So I removed the slack ID and secret from my docker-compose and restarted the container. The result is: only signing via email is available but not email gets sent out and after hitting the sign in button, I get redirected to the front-end of outline with no other reaction. 

If I use a random email address instead of using the one I previously used to sign in with slack as the admin user, I see a message: "a magic sign-in link has been sent if the account exists"

Was this supposed to work this way? 

SMTP is set up correctly, as when I first logged in via slack, I got a welcome email from the configured SMTP server.

**Reply by ovizii** [reply]
It turns out, if a user has used SSO, then for this account you can no longer use email sign-in :-( 
https://github.com/outline/outline/discussions/3660

**Reply by smit-io** [reply]
So, the only way to use magic link is to first use one of the SSO providers to log in. This is just crazy, especially when the documentation says this: "Email authentication is enabled by default if you have correctly setup an [SMTP provider]"

These needs to be a footnote that the first time login needs to be from an SSO.
